### PR TITLE
add atomic header to compile and add vscode support

### PR DESCRIPTION
**Pull request by li-chi**
1. request_pending_ with type atomic is introduced in PR#134, code did not compile (P469004206) for me. Added atomic header to fix the problem.
2. Added vscode support in .gitignore file.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,6 +36,7 @@ README.html
 *.swp
 *.autosave
 *~
+compile_commands.json
```
Comment (bhatvinay) [inline .gitignore:39]: Added by vscode?

**Review by bhatvinay — APPROVED** [review]
Make sure that you can build tp2 (do a local tp2_checkout and tp2_build)

File: .gitignore (hunk 1 above)
Comment (li-chi) [reply .gitignore:39]: After setting a CMake flag, it will be automatically generated during build. It is used by vscode for non-buck cpp project.

**Comment by li-chi** [comment]
> Make sure that you can build tp2 (do a local tp2_checkout and tp2_build)

Port change to tp2 and built sucessfully.
```
Successfully built kuduraft/1.8.raft/platform009
All projects built successfully.
```

**Pull request merged**